packages/vscode-extension/src/webview/views/PreviewView.tsx

```diff
@@ -295,17 +307,31 @@ function PreviewView() {
               setInspectFrame(item.frame);
             }
           }}
-          onCancel={() => resetInspector()}
+          onCancel={() => {
+            setInspectorState("disabled");
+          }}
         />
       )}
 
       <div className="button-group-bottom">
         <IconButton
-          active={isInspecting}
+          active={inspectorState !== "disabled"}
           tooltip={{
             label: "Select an element to inspect it",
           }}
-          onClick={() => setIsInspecting(!isInspecting)}
+          onClick={() => {
+            switch (inspectorState) {
+              case "disabled":
+                setInspectorState("inspecting");
+                break;
+              case "inspecting":
+                setInspectorState("disabled");
+                break;
+              case "selected":
+                setInspectorState("disabled");
+                break;
+            }
```

Comment:
I'd move this to a function (like `toggleInspecting`) to better communicate the intent of what this is supposed to do.